src/dev/flang/be/jvm/classfile/Expr.java

```diff
@@ -355,7 +685,7 @@ public void code(ClassFile.ByteCodeWriter ba, ClassFile cf)
    * create invokevirtual bytecode to call given class, name and descr producing
    * given result type on the stack.
    */
-  public static Expr invokeVirtual(String cls, String name, String descr, JavaType rt)
+  public static Expr invokeVirtual(String cls, String name, String descr, JavaType rt, int argCount)
```

Comment:
Instead of adding `argCount` as an argument here, try `ClassFileConstants.slotCountForArgs(descr)`.